### DISCUSSION TITLE
Vscode extension for paparazzi

### DISCUSSION THREAD

**Discussion by Fabien-B** [paparazzi/paparazzi#3324] [Show and tell]
Hi, I made a vscode extension for paparazzi. It's not doing much for now, but all ideas and contributions are welcome!
https://marketplace.visualstudio.com/items?itemName=paparazzi-uav.paparazzi-uav

It adds a button in the status bar to select an AC and a target, then update the paparazzi tab in the left panel with the configuration files for this AC and the modules that are loaded (explicitly or by dependencies).

It also list all paparazzi modules.

![image](https://github.com/paparazzi/paparazzi/assets/7016951/a5828f81-c8f8-4c84-8d83-f2b29167b595)

**Comment by Fabien-B** [comment]
Maybe it can interest you @FlorianSan @fvantienen @dewagter

**Comment by dewagter** [comment]
@Fabien-B really nice idea. I installed it and can see the paparazzi icon on the left. It does not fill the list of modules and AC_conf yet so I might need to configure something differently.

In any case there are plenty of things a good plugin could help with. From checking for every module that all defines are in the documentation and are still relevant, to inserting all the defines when adding a module, to standardizing the file layout and sequence so they can better be diffed, to checking which airframes XML files are in no single conf (and thus not test-compiled)...
Check ```./sw/tools/airframe_editor/``` to see previous efforts.

**Reply by Fabien-B** [reply]
Did you opened the paparazzi directory ? For now it must be the only folder in the workspace for the extension to work (I will try to improve that later). It should then read `conf/conf.xml` to get the aircrafts.
It should then show a notification: `Paparazzi update successfull!`

**Comment by fvantienen** [comment]
Will try it out :)
